src/avalanchemq/http/handler/basic_auth.cr

```diff
@@ -58,6 +58,11 @@ module AvalancheMQ
         return false if user.password.to_s.empty?
         user.password.not_nil!.verify(password)
       end
+
+      private def guest_localhost?(context, user)
+        return true unless user.name == "guest"
+        context.request.remote_address.as(Socket::IPAddress).loopback?
```

Comment:
can "RemoteAddress" be somthing else? could this raise an exception at some point?

Reply:
HTTP Server binds to TCP, TCP with TLS or unix socket. So unless a family check is done I assume it could fail with Socket::IPAddress cast if the connection is done through the unix socket. Also isn't the connection done through a unix socket always on the same host?